client/src/components/river/index.tsx

```diff
@@ -155,11 +186,36 @@ const FhirRiverView = (): React.ReactElement => {
             disabled={!selectedSource.id || credentialsMissing}
             loading={running}
             className="button-submit"
-            onClick={onClickRun}
+            onClick={onClickCreateBatch}
           >
             Make the river flow
           </Button>
         </div>
+        <StringSelect
+          items={Object.keys(batchList).map(
+            (batchId: string) => `${batchId} ${batchList[batchId].timestamp}`
+          )}
+          inputItem={
+            selectedBatch
+              ? `${selectedBatch.id} ${selectedBatch.timestamp}`
+              : 'Select batch to cancel'
+          }
+          onChange={(item: string): void =>
+            setSelectedBatch({
+              id: item.split(' ')[0],
+              timestamp: item.split(' ')[1]
```

Comment:
lol this is pretty hacky to say the least. I'm ok to leave it like this since this pyrog won't last very long, but it would have been cleaner to use only the timestamp in this select, or write a BatchSelect.

Reply:
The client will survive, right? 👀